### PR TITLE
[risk=low][no ticket] Remove all code related to enableResearchPurposePrompt and needsReviewPrompt

### PR DESCRIPTION
**Pull request by jmthibault79**
After a release cycle, we can fully remove it: https://github.com/all-of-us/workbench/pull/7875

Tested locally by making Research Purpose modifications without problems and uploading a reporting snapshot

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [x] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [x] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [x] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/api/OfflineWorkspaceController.java [hunk 1]
```diff
@@ -1,70 +0,0 @@
...
-import javax.inject.Provider;
-import org.pmiops.workbench.config.WorkbenchConfig;
-import org.pmiops.workbench.db.dao.WorkspaceDao;
-import org.pmiops.workbench.db.model.DbWorkspace;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.RestController;
-
-@RestController
-public class OfflineWorkspaceController implements OfflineWorkspaceApiDelegate {
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/OfflineWorkspaceController.java:16]: "offline" controllers are only called by crons, not the API.  So it's safe to delete them without regard to UI deployment.

File: ui/src/app/pages/workspace/workspace-nav-bar.tsx [hunk 2]
```diff
@@ -140,7 +140,11 @@ const CdrVersion = (props: {
   );
 };
 
-const tabs = [
+interface Tab {
+  name: string;
+  link: string;
+}
+const tabs: Tab[] = [
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-nav-bar.tsx:147]: unrelated

File: api/src/main/java/org/pmiops/workbench/db/dao/WorkspaceDao.java [hunk 3]
```diff
@@ -116,10 +116,6 @@ Optional<DbWorkspace> findFirstByGoogleProjectAndActiveStatusOrderByLastModified
 
   Set<DbWorkspace> findAllByCreator(DbUser user);
 
-  Set<DbWorkspace> findAllByActiveStatus(short activeStatus);
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/db/dao/WorkspaceDao.java:119]: removed a few unrelated unused methods too

**Review by Peter-Lavigne — APPROVED** [review]
LGTM. Tested by modifying research purpose and uploading a reporting snapshot.

Out of curiosity I looked into why we added this in the first place. I hate to do this to you again so soon, but this was backed by a CAPS requirement. The [story for the prompt](https://precisionmedicineinitiative.atlassian.net/browse/RW-2400) links to a [design doc](https://docs.google.com/document/d/1aINtt2IQtRzQB8pvwA2hly49JPtPg_L1FBuHWrT-QQk/edit#heading=h.vrvyffcte9tr) (@NehaBroad's!) which links to [a doc](https://docs.google.com/document/d/10lFDOWGB3FlTONCzZNK1yQzfC65zl2o2Ul0ugyunEIw/edit#heading=h.4zq82h8vezj0) which claims this is a CAPS requirement. Looking at the [CAPS doc](https://docs.google.com/document/d/1jK66dh7qP9VOL3e4GoFATFZMLvgBl81hvvTL6WKklYo/edit), "Users will be prompted every 365 days after the Workspace creation to update the Workpsace descriptions or confirm that there are no updates."

I'm still approving this since this change reflects no workbench changes - we never rolled out those requirements in the first place. I think it's worth investigating a bit further while we have context on CAPS. If it's been this long without implementing it I suspect we don't need it.

I also think worthwhile to drop the DB columns and update the WRD column description but no need to do that in this PR.

Why was [this PR](https://github.com/all-of-us/workbench/pull/7875) split out?

**Comment by jmthibault79** [comment]
> If it's been this long without implementing it I suspect we don't need it.

I agree, but this is certainly not *urgent* so I'm ok waiting a bit on confirmation before merging.

> I also think worthwhile to drop the DB columns

Yes, that's a future PR.  Not doing it here out of caution regarding deployment safety.  We should only remove DB fields after all environments are free of code which references them, and enough time has passed that we're confident we don't need to roll back the code changes.

> update the WRD column description but no need to do that in this PR.

Can't do that in this repo :)

> Why was https://github.com/all-of-us/workbench/pull/7875 split out?

Also due to deployment safety: we don't deploy the API and UI at exactly the same time.  In the best case, we have ~ 10 min where the new API needs to support the old UI before it is also updated.  In the worst case, something delays the release and they're out-of-sync for longer.

**Comment by Peter-Lavigne** [comment]
That all makes sense, thanks!
